src/Screens/CartScreen/index.tsx

```diff
@@ -0,0 +1,113 @@
+import { useState, useEffect } from "react";
+import { useSelector, useDispatch } from "react-redux";
+import { UserStatusInterface } from "../../Components/Navbar/NavbarInterface";
+import { db } from "../../Firebase";
+import { collection, doc, getDoc } from "firebase/firestore";
+import { infoDataType } from "../Home/HomeInterface";
+import {
+  CartPageContainer,
+  CartItemsContainer,
+  CartSummaryContainer,
+  CartSummaryTitle,
+  CartSummaryItem,
+  CartTotalPrice,
+  CartEmptyMessage,
+} from './CartScreenStyle';
+import { CartCard } from "../../Components/CartCard";
+import { CartItemState } from "../../Redux/Reducer/SetCartItems";
+import { saveCartItems } from "../../Redux/Actions";
+
+
+export const CartScreen = () => {
+  const dispatch = useDispatch();
+  const cartItems = useSelector((state: CartItemState) => state.cartItem);
+  const getTotalPrice = () => {
+    return cartItems.reduce(
+      (total, item) => total + item.price * item.quantity,
+      0
+    );
+  };
+
+  const discountedPrice = Math.floor(getTotalPrice() * 0.1);
```

Comment:
Be consistent, keep variables, states at top and then functions as it helps in understanding things easily